### DISCUSSION TITLE
Multipart file upload

### DISCUSSION THREAD

**Discussion by jessepinkman9900** [Netflix/dgs-framework#351] [Q&A] [answered]
- tried using `scalar Upload` from https://netflix.github.io/dgs/advanced/file-uploads/ 
- getting build failure due to 
  ```
  error: cannot find symbol
   private Upload file;
  ```

- using dgs platform bom `3.11.0`
- is there any way to register MultipartFile as custom scalar ?
  - defining `scalar MultiparFile`
  - defining typeMapping like 
    ```
     typeMapping = ["MultipartFile": "org.springframework.web.multipart.MultipartFile"]
    ```  
   - then register scalar using `@DgsComponent` and `@DgsRuntimeWiring`
- looking for example code or bug fix

**Comment by srinivasankavitha** [comment, marked as answer]
You can't have `Upload` in your code, that is for declaring in your schema. So in your schema, you would have:
```
scalar Upload
type someType {
    Upload someFile
}
or
type Mutation {
    uploadFile(file: Upload)
}
```

The scalar `Upload` is already available as part of the framework.

Here is an example: https://github.com/Netflix/dgs-examples-java/blob/main/src/main/java/com/example/demo/datafetchers/ArtworkUploadDataFetcher.java
Schema: 
https://github.com/Netflix/dgs-examples-java/blob/main/src/main/resources/schema/schema.graphqls#L8

**Reply by jessepinkman9900** [reply]
declared it in schema, error was thrown in codegen. `@skipcodegen` did the trick. thank you!

**Reply by srinivasankavitha** [reply]
Ah ok, yes we don't have codegen working with this for file uploads.

**Reply by victorlevasseur** [reply]
We used this snippet to properly support Upload scalar with the codegen:
```groovy
generateJava {
	// other config...
	typeMapping = ["Upload": "org.springframework.web.multipart.MultipartFile"]
}
```

**Reply by jessepinkman9900** [reply]
thank you @victorlevasseur
